File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentSource.java
package org.wordpress.android.ui.comments;

import org.wordpress.android.util.analytics.AnalyticsUtils.AnalyticsCommentActionSource;

enum CommentSource {
    NOTIFICATION,
    SITE_COMMENTS;

    AnalyticsCommentActionSource toAnalyticsCommentActionSource() {
        switch (this) {
            case NOTIFICATION:
                return AnalyticsCommentActionSource.NOTIFICATIONS;
            case SITE_COMMENTS:
                return AnalyticsCommentActionSource.SITE_COMMENTS;
        }
        throw new IllegalArgumentException(
                this + " CommentSource is not mapped to corresponding AnalyticsCommentActionSource");
    }
}

File: WordPress/src/main/java/org/wordpress/android/ui/comments/unified/CommentSource.kt
package org.wordpress.android.ui.comments.unified

import org.wordpress.android.util.analytics.AnalyticsUtils.AnalyticsCommentActionSource

enum class CommentSource {
    NOTIFICATION, SITE_COMMENTS;

    fun toAnalyticsCommentActionSource(): AnalyticsCommentActionSource =
        when (this) {
            NOTIFICATION -> AnalyticsCommentActionSource.NOTIFICATIONS
            SITE_COMMENTS -> AnalyticsCommentActionSource.SITE_COMMENTS
        }
}
